### DISCUSSION TITLE
Issue with `\ref`  command in math environment

### DISCUSSION THREAD

**Discussion by anontokyo2077** [jgm/pandoc#10377] [Q&A] [answered]
When I'm trying to convert my tex work to a docx file，the command `\ref` seems to cause an issue when used inside an `align`.
Here's the simplified latex content:
```
\begin{align}
\ref{table: graduates}
\end{align}
```
and the warning is:
```
[WARNING] Could not convert TeX math \begin{aligned}
  \ref{table: graduates}
  \end{aligned}, rendering as TeX:
  \ref{table: graduates}
      ^
  unexpected control sequence \ref
  expecting "%", "\\label", "\\tag", "\\nonumber", whitespace or "\\allowbreak"
```
How can I use the `\ref` command sucessfully in the `align` environment? Thanks!

**Comment by jgm** [comment, marked as answer]
We don't support `\ref` in math yet, sorry.  It's difficult to see how to support this. Math conversion is done by `texmath`, and it wouldn't know how to resolve the reference.  Perhaps pandoc could do its own macro substitution and resolve the reference before passing the tex to `texmath`, but currently there is no tracking of equation numbers.

As a workaround, you might add this to the beginning of your tex file:
```
\renewcommand{\ref}[1]{?}
```
This will allow the math to be converted without error, but all references (including those outside of math) will be `?`.  (Currently pandoc will resolve references to sections and figures outside of math; if you want to keep this you might need to use etoolbox and do something so that the `\ref` command is only redefined inside math.)

**Reply by anontokyo2077** [reply]
Thanks for your help！
